dokan_fuse/src/dokanfuse.cpp

```diff
@@ -433,6 +433,87 @@ static NTSTATUS DOKAN_CALLBACK FuseUnmounted(PDOKAN_FILE_INFO DokanFileInfo) {
   return errno_to_ntstatus_error(impl->unmounted(DokanFileInfo));
 }
 
+static DWORD AddSeSecurityNamePrivilege() {
+    LUID luid;
+    if (!LookupPrivilegeValue(0, SE_SECURITY_NAME, &luid)) {
+        DWORD err = GetLastError();
+        if (err != ERROR_SUCCESS) {
+            return err;
+        }
+    }
+
+    HANDLE token = 0;
+    if (!OpenProcessToken(GetCurrentProcess(),
+        TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY, &token)) {
+        return GetLastError();
+    }
+
+    ON_BLOCK_EXIT([token] {
+        CloseHandle(token);
+    });
+
+    LUID_AND_ATTRIBUTES attr;
+    attr.Attributes = SE_PRIVILEGE_ENABLED;
+    attr.Luid = luid;
+
+    TOKEN_PRIVILEGES priv;
+    priv.PrivilegeCount = 1;
+    priv.Privileges[0] = attr;
+
+    AdjustTokenPrivileges(token, FALSE, &priv, NULL, NULL, NULL);
+    DWORD err = GetLastError();
+    if (err != ERROR_SUCCESS) {
+        return err;
+    }
+
+    return ERROR_SUCCESS;
+}
+
+static NTSTATUS DOKAN_CALLBACK FuseGetFileSecurity(
+    LPCWSTR FileName, PSECURITY_INFORMATION SecurityInformation,
+    PSECURITY_DESCRIPTOR SecurityDescriptor, ULONG BufferLength,
+    PULONG LengthNeeded, PDOKAN_FILE_INFO DokanFileInfo) {
+  impl_fuse_context *impl = the_impl;
+  if (impl->debug())
+    FWPRINTF(stderr, L"GetFileSecurity %x\n", *SecurityInformation);
+
+  bool needASS = ((*SecurityInformation & SACL_SECURITY_INFORMATION) ||
+      (*SecurityInformation & BACKUP_SECURITY_INFORMATION));
+
+  if (needASS) {
+      DWORD result = AddSeSecurityNamePrivilege();
+      if (ERROR_SUCCESS != result)
+          return lasterror_to_ntstatus(result);
+  }
+
+  HANDLE handle = CreateFileW(
+      DokanFileInfo->DokanOptions->MountPoint,
```

Comment:
@matepek Oh this seems to be a dirty hack :smile: I am not sure the system is not going to like to get everytime the same security informations.

Your implementation would be the same as returning virtual static security information since `DokanFileInfo->DokanOptions->MountPoint` as been created by dokan.

I understand why you did this since you cannot request the informations to the file system or by the fuse API.
But I am sceptical on this implementation :cry: